### ISSUE TITLE
Autoformatting messes up indentation when starting a new line

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#73921]
Manually moved here from AB#2078411.

Repro'd in Version 17.11.0 Preview 2.0 [34930.64.main]
Start with this code:

    public class CommandList
    {
        private readonly IReadOnlyList<string> _commands;
        public CommandList(params string[] commands)
        {
            _commands = commands;
        }
        private static readonly Dictionary<string, Commands> AllCommands = new Dictionary<string, Commands>()
                                                                                    {
                                                                                        { "Help", Commands.Help },
                                                                                        { "Retry", Commands.Retry },
                                                                                        { "Undo", Commands.Undo },
                                                                                        { "Redo", Commands.Redo },

from the end of the last line and hit <kbd>Enter</kbd> and you get:

![image](https://github.com/dotnet/roslyn/assets/1408396/98fdc3f4-8090-4fc0-9526-fda66f375034)

(note the caret is properly indented via the smart indentation service). Type <kbd>{</kbd> and get:

![image](https://github.com/dotnet/roslyn/assets/1408396/699442e1-4161-4770-8bcb-440f7e360dbe)

and an annoyed user.

At the very least, any type of formatting that adjusts the indentation should use the same logic the `ISmartIndentation` service uses to determine the indentation. Having two pieces of code that are fighting over the proper indentation is not good.

**Comment by CyrusNajmabadi** [comment]
> Having two pieces of code that are fighting over the proper indentation is not good.

Agreed.  The indentation/formatting split is a nightmare that continues to plague us.  Some parts are shared.  but it appears as if *each* has codepaths that the other doesn't go through.   It's not good.

**Comment by ToddGrun** [comment]
@sharwell / @CyrusNajmabadi -- Trying to triage this, and yeah, this definitely isn't great. Is there someone on the team who usually gets either formatting or smart indent bugs?

**Comment by CyrusNajmabadi** [comment]
Sam does :-). Or round robin it

**Comment by ToddGrun** [comment]
Given the disparity and complexity of both systems, making them match better feels like more than a bug level fix to me. Setting the feature label so we can address this as such.

**Comment by sharwell** [comment]
Well, you definitely don't want the formatter to handle this indentation case because the formatter doesn't ever do anything inside this collection initializer.
